cli/index.ts

```diff
@@ -229,11 +238,18 @@ async function cli(args: CommandLineArgs): Promise<Options | null> {
 
   const { appOptions, outputDir } = await promptOptions(commandLineOptions);
 
+  if (appOptions.backend === "python" && appOptions.api === "graphql") {
```

Comment:
I think it'd be better to error out earlier, we can create add a `validateCommandLineOptions` function that will be called right after parsing command line args, and maintain a deny list of options for extensibility:

```ts
const OPTION_COMBINATION_DENY_LIST = [
  [
    { optionType: "backend", value: "python" },
    { optionType: "api", value: "graphql" },
  ],
];

const validateCommandLineOptions = (
  commandLineOptions: CommandLineOptions,
): void => {
  OPTION_COMBINATION_DENY_LIST.forEach((combination) => {
    if (
      combination.every(
        (option) =>
          commandLineOptions[option.optionType as keyof CommandLineOptions] ===
          option.value,
      )
    ) {
      // TODO: custom error type would be a nice-to-have
      throw new Error(`${JSON.stringify(combination)} is not a valid combination`);
    }
  });
};
```